packages/devtools_app/lib/src/standalone_ui/ide_shared/property_editor/property_editor_view.dart

```diff
@@ -173,27 +184,290 @@ class _PropertyInput extends StatelessWidget {
+    extends State<_ExpandableWidgetDocumentation>
+    with SingleTickerProviderStateMixin {
+  bool _isExpanded = false;
+  late AnimationController _expandAnimationController;
+  late Animation<double> _expandAnimation;
+
+  @override
+  void initState() {
+    super.initState();
+    _expandAnimationController = defaultAnimationController(this);
+    _expandAnimation = defaultCurvedAnimation(_expandAnimationController);
+  }
+
+  @override
+  void dispose() {
+    _expandAnimationController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    final regularFontStyle = theme.regularTextStyle;
+    final fixedFontStyle = theme.fixedFontStyle.copyWith(
+      color: theme.colorScheme.primary,
+    );
+    final paragraphs = widget.documentation.split('\n');
+
+    if (paragraphs.length == 1) {
+      return RichText(
+        text: _convertDartDoc(
+          widget.documentation,
+          regularFontStyle: regularFontStyle,
+          fixedFontStyle: fixedFontStyle,
+        ),
+      );
+    }
+
+    final [firstParagraph, ...otherParagraphs] = paragraphs;
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        // Currently this will process the Dart doc each time a user expands
+        // or collapses the text block. Because the Dart doc is never very
+        // large, this is not an expensive operation. However, we could
+        // consider caching the result if this needs to be optimized.
+        RichText(
+          text: _convertDartDoc(
+            firstParagraph,
+            regularFontStyle: regularFontStyle,
+            fixedFontStyle: fixedFontStyle,
+          ),
+        ),
+        if (_isExpanded)
+          FadeTransition(
+            opacity: _expandAnimation,
+            child: RichText(
+              text: _convertDartDoc(
+                otherParagraphs.join('\n'),
+                regularFontStyle: regularFontStyle,
+                fixedFontStyle: fixedFontStyle,
+              ),
+            ),
+          ),
+        GestureDetector(
+          onTap: _toggleExpansion,
+          child: Padding(
+            padding: const EdgeInsets.only(top: denseSpacing),
+            child: Text(
+              _isExpanded ? 'Show less' : 'Show more',
+              style: theme.boldTextStyle.copyWith(
+                color: theme.colorScheme.primary,
+              ),
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+
+  void _toggleExpansion() {
+    setState(() {
+      _isExpanded = !_isExpanded;
+      if (_isExpanded) {
+        _expandAnimationController.forward();
+      } else {
+        _expandAnimationController.reverse();
+      }
+    });
+  }
+
+  /// Converts a [dartDocText] String into a [TextSpan].
+  ///
+  /// Removes any brackets and backticks and displays the text inside them as
+  /// fixed font.
+  TextSpan _convertDartDoc(
```

Comment:
can you make this visible for testing and add tests for this logic?

Reply:
Yes! Moved this to a `utils` file instead though since it doesn't need to be part of this widget (and it makes it easier to test)